### PR TITLE
topn:fix bug in order by primary key

### PR DESCRIPTION
**Pull request by AndreMouche**
Hi,all,
     this PR is used to fix bug in order by primary key.

@disksing @BusyJay PTAL

### CODE REVIEW

**Comment by ngaut** [comment]
failures:
    storage::test_storage::test_txn_store_gc2_6

**Comment by AndreMouche** [comment]
This bug would be fixed in [PR#1622](https://github.com/pingcap/tikv/pull/1622) @ngaut

**Comment by ngaut** [comment]
LGTM. PTAL @hanfei1991

File: tests/coprocessor/test_select.rs [hunk 1]
```diff
@@ -893,6 +893,37 @@ fn test_order_by_column() {
...
 #[test]
+fn test_order_by_pk() {
+    let mut data = vec![
+    (8, Some("name:0"), 2),
+    (7, Some("name:3"), 3),
+    (6, Some("name:0"), 1),
+    (5, Some("name:6"), 4),
+    (4, Some("name:5"), 4),
+    (3, Some("name:4"), 4),
+    (2, None, 4),
```
Comment (disksing) [inline tests/coprocessor/test_select.rs]: Looks not well formatted. The rest LGTM.

**Comment by shenli** [comment]
@disksing PTAL

**Comment by disksing** [comment]
LGTM

File: tests/coprocessor/test_select.rs [hunk 2]
```diff
@@ -893,6 +893,37 @@ fn test_order_by_column() {
...
+        (4, Some("name:5"), 4),
+        (3, Some("name:4"), 4),
+        (2, None, 4),
+    ];
+
+
+    let product = ProductTable::new();
+    let (_, mut end_point) = init_with_data(&product, &data);
+    let req = Select::from(&product.table)
+        .order_by(product.id, true)
```
Comment (BusyJay) [inline tests/coprocessor/test_select.rs]: I don't think this case is related to this fix. If you remove the fix, can this case pass?

**Comment by hanfei1991** [comment]
LGTM

**Comment by BusyJay** [comment]
LGTM

**Pull request merged**